include/jemalloc/internal/hpa_opts.h

```diff
@@ -59,6 +59,36 @@ struct hpa_shard_opts_s {
 	 * Maximum number of hugepages to purge on each purging attempt.
 	 */
 	ssize_t experimental_max_purge_nhp;
+
+	/*
+	 * When the number of inactive bytes in a hugepage is >=
+	 * purge_threshold, the page is purgable.  Setting this to HUGEPAGE
+	 * would only purge empty pages if start_as_huge is true and the purges
+	 * would be exactly HUGEPAGE bytes.  Depending on your kernel settings
+	 * purging from non-empty hugepage may result in loss of performance.
+	 */
+	size_t purge_threshold;
+
+	/*
+	 * Minimum number of ms that needs to elapse between HP page becoming
+	 * eligible for purging and actually getting purged.  Setting this to
+	 * a larger number would give better chance of reusing that memory.
+	 * Setting it to 0 means that page is eligible for purging as soon as
+	 * it meets the purge_threshold.  The clock resets when purgability of
+	 * the page changes.
+	 */
+	uint64_t min_purge_delay_ms;
+
+	/*
+	 * If page should start as huge (instead of waiting to for hugification
+	 * threshold to be reached).  This allows us to utilize HP immediately
+	 * and have similar behavior whether the thp setting is 'always' or
+	 * 'madvise'.  When using this option you probably want to purge less
+	 * aggressively: either no purge at all (dirty_mult=-1), or purge only
+	 * empty pages (purge_threshold=HUGEPAGE) with some delay that allows
+	 * their reuse (for example the period between memory peaks).
+	 */
+	bool start_as_huge;
```

Comment:
Are all three options here meant to be long term? Or they are parts of an initial attempt to adapt better to thp always and some of these options will be gone? Asking because I am trying to evaluate if we want to add `experimental` here.